bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/storage/Storage.java

```diff
@@ -41,6 +41,16 @@
     T remove(@NonNull String key);
 
     /**
+     * Check if the storage contains a key.
+     *
+     * @param key the key
+     * @return true if the storage contains the key, otherwise false
+     */
+    default boolean containsKey(@NonNull String key) {
+        return getKeys().contains(key);
```

Comment:
I have added a default implementation to not break any API -- if you use it in your custom products.
Perhaps it would be better to drop that default implementation to force the implementer to use a better / custom implementation. 
WDYT?

Reply:
We have our own implementation, but that would be easily manageable. So from my side it would be okay to drop it and break the API in that place.

Reply:
@kaikreuzer Okay for you to drop the default implementation and risk an API break?

Reply:
Yes, ok for me!